adios2: Update to 2.10.1

martin-g opened the pull request.

martin-g committed:
adios2: Update to 2.10.1

Change PIC value to TRUE
As at https://github.com/ornladios/ADIOS2/tree/master/scripts/build_scripts

Explicitly pass CMAKE_**_COMPILER options

Remove unused options

```
CMake Warning:
2024-10-10T11:52:05.9700227Z   Manually-specified variables were not used by the project:
2024-10-10T11:52:05.9700853Z
2024-10-10T11:52:05.9701038Z     CMAKE_FC_COMPILER
2024-10-10T11:52:05.9701539Z     CMAKE_SKIP_INSTALL_RPATH
```

And do not use `-DCMAKE_SKIP_RPATH:BOOL=ON` since it leads to build
errors

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

adrianreber commented:
One of the tests dumps, core. Hmm. Is it a broken test or an upstream regression?

martin-g committed:
Try with ReadRandomAccess mode when reading a file

Current error:
```
Log file: ./tests/libs/adios2/tests/family-gnu14-mpich/arrays_read.log
2024-10-10T13:31:18.3670468Z terminate called after throwing an instance of 'std::logic_error'
2024-10-10T13:31:18.3676365Z   what():  ␛[1;36m[Thu Oct 10 13:30:59 2024]␛[1;34m [ADIOS2 EXCEPTION]␛[0m <Engine> <BP5Reader> <NotifyEngineNoVarsQuery> : You've called InquireVariable() when the IO is empty and outside a BeginStep/EndStep pair.  If this is code that is newly transititioning to the BP5 file engine, you may be relying upon deprecated behaviour.  If you intend to use ADIOS using the Begin/EndStep interface, move all InquireVariable calls inside the BeginStep/EndStep pair.  If intending to use random-access file mode, change your Open() mode parameter to Mode::ReadRandomAccess.␛[0m
```

Signed-off-by: Martin Tzvetanov Grigorov <mgrigorov@apache.org>

martin-g commented:
Fixed it!
